src/de/mod.rs

```diff
@@ -483,6 +483,9 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         self.bytes.skip_ws()?;
 
         if self.bytes.consume("(") {
+            if self.bytes.exts.contains(Extensions::IMPLICIT_OUTMOST_STRUCT) {
```

Comment:
We are in a general struct deserialization routine here. If we error whenever `IMPLICIT_OUTMOST_STRUCT` is enabled, we wouldn't be able to deserialize any non-root struct. Correct?